File: 1062-longest-repeating-substring/1062-longest-repeating-substring.cpp
class Solution {
public:
    int longestRepeatingSubstring(string s) {
        int n = s.length();
        for(int w = n-1;w>=1;w--){
            int j = 0;
            set<string> m;
            while(j <= n-w){
                string t = s.substr(j,w);
                if(m.find(t) != m.end()){
                    return t.length();
                }
                else{
                    m.insert(t);
                }
                j++;
            }
        }
        return 0;
    }
};

File: 1062-longest-repeating-substring/1062-longest-repeating-substring.py
class Solution:
    def longestRepeatingSubstring(self, s: str) -> int:
        n = len(s)
        suf = [s[i:] for i in range(n)]
        suf.sort()
        c = 0
        for i in range(1,n):
            j = 0
            while(j< min(len(suf[i]), len(suf[i-1])) and suf[i][j] == suf[i-1][j]):
                j+=1
            c = max(c, j)
        return c